warehouse/query-core/src/main/java/datawave/next/DocIdIterator.java

```diff
@@ -0,0 +1,222 @@
+package datawave.next;
+
+import java.util.Collections;
+
+import org.apache.accumulo.core.data.ArrayByteSequence;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Range;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.iterators.SortedKeyValueIterator;
+import org.apache.commons.jexl3.parser.ASTEQNode;
+import org.apache.commons.jexl3.parser.JexlNode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Preconditions;
+
+import datawave.query.jexl.JexlASTHelper;
+import datawave.query.jexl.visitors.JexlStringBuildingVisitor;
+
+/**
+ * A document id iterator that supports simple equality terms
+ */
+public class DocIdIterator extends BaseDocIdIterator {
+
+    private final static Logger log = LoggerFactory.getLogger(DocIdIterator.class);
+
+    protected String value;
+
+    /**
+     * Default constructor
+     *
+     * @param source
+     *            the source
+     * @param row
+     *            the row
+     * @param node
+     *            an EQ node
+     */
+    public DocIdIterator(SortedKeyValueIterator<Key,Value> source, String row, JexlNode node) {
+        super(source, row, node);
+        parseNode(node);
+        this.prefix = "fi\0" + field;
+        this.columnFamilies = Collections.singleton(new ArrayByteSequence(prefix));
+    }
+
+    /**
+     * Parses the JexlNode, assumed to be a {@link ASTEQNode}.
+     *
+     * @param node
+     *            a JexlNode
+     */
+    protected void parseNode(JexlNode node) {
+        this.field = JexlASTHelper.getIdentifier(node);
+        Preconditions.checkNotNull(field, "Expected field to not be null for node: " + JexlStringBuildingVisitor.buildQuery(node));
+
+        Object literal = JexlASTHelper.getLiteralValue(node);
+        Preconditions.checkNotNull(literal, "Expected value to not be null for node: " + JexlStringBuildingVisitor.buildQuery(node));
+        this.value = String.valueOf(literal);
+    }
+
+    @Override
+    public boolean hasNext() {
+        if (!seenSeek) {
+            handleFirstSeek();
+        }
+
+        while (tk == null && source.hasTop()) {
+            Key top = source.getTopKey();
+            parser.parse(top);
+
+            // apply time and datatype filter
+            if (!acceptDatatype(parser.getDatatype())) {
+                handleDatatypeMiss();
+                continue;
```

Comment:
```suggestion
                handleDatatypeMiss();
                safeNext();
                continue;
```

Reply:
I will look at this

Reply:
General cleanup here https://github.com/NationalSecurityAgency/datawave/issues/2916